utils/utils.py

```diff
@@ -18,14 +18,59 @@
 )
 from utils.visualizer import Visualizer
 
+import re
+
+
+def parse_slurm_nodelist(nodelist):
```

Comment:
This seems like something we shouldn't need to do, but I'm glad that you know all these details and fixed it for us